packages/esm-patient-vitals-app/src/components/delete-vitals-biometrics-modal/delete-vitals-biometrics-modal.component.tsx

```diff
@@ -0,0 +1,125 @@
+}) => {
+  const { t } = useTranslation();
+  const { mutate } = useVitalsAndBiometrics(patientUuid, formType);
+  const [isDeleting, setIsDeleting] = useState(false);
+  const targetConcepts = useVitalsOrBiometricsConcepts(formType);
+  const {
+    encounter,
+    isLoading: isLoadingEncounter,
+    mutate: mutateEncounter,
+  } = useEncounterVitalsAndBiometrics(encounterUuid);
+
+  const messages = useMemo(
+    () => ({
+      success: {
+        vitals: t('vitalsDeleted', 'Vitals deleted'),
+        biometrics: t('biometricsDeleted', 'Biometrics deleted'),
+      },
+      error: {
+        vitals: t('errorDeletingVitals', 'Error deleting vitals'),
+        biometrics: t('errorDeletingBiometrics', 'Error deleting biometrics'),
+      },
+      modalTitle: {
+        vitals: t('deletePatientVitals', 'Delete Vitals'),
+        biometrics: t('deletePatientBiometrics', 'Delete Biometrics'),
+      },
+      modalDescription: {
+        vitals: t('deleteVitalsConfirmationText', 'Are you sure you want to delete these Vitals entries?'),
+        biometrics: t('deleteBiometricsConfirmationText', 'Are you sure you want to delete these Biometrics entries?'),
+      },
+    }),
+    [t],
+  );
+
+  const handleDelete = useCallback(async () => {
+    setIsDeleting(true);
+
+    if (!isLoadingEncounter && encounter) {
+      const obsToVoid = encounter.obs
+        .filter((obs) => targetConcepts.includes(obs.concept.uuid))
+        .map((obs) => ({ uuid: obs.uuid, voided: true }));
+
+      createOrUpdateVitalsAndBiometrics(
+        patientUuid,
+        encounter.encounterType.uuid,
+        encounter.uuid,
+        null,
+        obsToVoid,
+        new AbortController(),
+      )
+        .then((res) => {
+          mutate();
+          mutateEncounter();
+          closeDeleteModal();
+          showSnackbar({
+            isLowContrast: true,
+            kind: 'success',
+            title: messages.success[formType],
+          });
+        })
+        .catch((error) => {
+          console.error('Error deleting biometrics: ', error);
+          showSnackbar({
+            isLowContrast: false,
+            kind: 'error',
+            title: messages.error[formType],
+            subtitle: error?.message,
+          });
+        })
+        .finally(() => setIsDeleting(false));
+    }
+  }, [
+    isLoadingEncounter,
+    encounter,
+    patientUuid,
+    targetConcepts,
+    messages.success,
+    messages.error,
+    formType,
+    closeDeleteModal,
+    mutate,
+    mutateEncounter,
+  ]);
+
+  return (
+    <>
+      <ModalHeader closeModal={closeDeleteModal} title={messages.modalTitle[formType]} />
+      <ModalBody>
+        <p>{messages.modalDescription[formType]}</p>
+      </ModalBody>
+      <ModalFooter>
+        <Button kind="secondary" onClick={closeDeleteModal}>
+          {t('cancel', 'Cancel')}
+        </Button>
+        <Button kind="danger" onClick={handleDelete} disabled={isDeleting}>
+          {t('delete', 'Delete')}
```

Comment:
Can we a loading state to this button (similar to [this](https://github.com/openmrs/openmrs-esm-patient-chart/blob/main/packages/esm-patient-conditions-app/src/conditions/delete-condition.modal.tsx#L55))?